### PR TITLE
[dynamo] Enable typechecking for testing.py

### PR DESCRIPTION
**Pull request by int3**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #112129
* #112128
* #112127
* #112031
* #111992
* #111894

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

### CODE REVIEW

File: torch/_dynamo/testing.py [hunk 1]
```diff
@@ -187,7 +188,7 @@ def __init__(self):
         self.frame_count = 0
         self.op_count = 0
 
-    def __call__(self, gm: torch.fx.GraphModule, example_inputs):
+    def __call__(self, gm: torch.fx.GraphModule):
```
Comment (Skylion007) [inline torch/_dynamo/testing.py:191]: Is this intentional? We aren't altering the testing behavior right?

Comment (eellison) [reply torch/_dynamo/testing.py:191]: Right, I think we need to keep this. They should be List[torch.Tensor]

File: torch/_dynamo/testing.py [hunk 2]
```diff
@@ -252,7 +253,7 @@ def standard_test(self, fn, nargs, expected_ops=None, expected_ops_dynamic=None)
         expected = CompileCounter()
         try:
             gm = torch.fx.symbolic_trace(fn)
-            expected(gm)
+            expected(gm, [])
```
Comment (int3) [inline torch/_dynamo/testing.py:256]: I removed the param earlier, but I guess the right fix is to add the missing argument here

Comment (int3) [reply torch/_dynamo/testing.py]: Ended up using a type-igore/FIXME instead. Details: https://github.com/pytorch/pytorch/issues/112230

**Comment by Skylion007** [comment]
Ah, please fix that type ignore/ issue either in this PR or a separate oneS It does appear to be an actual bug after looking at that linked issue. Otherwise this PR looks good to me.

**Review by Skylion007 — APPROVED** [review]

**Comment by int3** [comment]
I don't really have context on that issue, that's why I created the task -- I'm hoping someone else might pick it up. My plan is to finish the typing of all the dynamo files first before revisiting these less-obvious issues.

**Pull request closed without merging**